migration/pkg.go

```diff
@@ -0,0 +1,111 @@
+package migration
+
+import (
+	"encoding/json"
+	"fmt"
+	liberr "github.com/konveyor/controller/pkg/error"
+	"github.com/konveyor/controller/pkg/logging"
+	"github.com/konveyor/tackle2-hub/database"
+	"github.com/konveyor/tackle2-hub/model"
+	"gorm.io/gorm"
+)
+
+var log = logging.WithName("migration")
+
+const HubDBVersion = ".hub.db.version"
+
+//
+// Migration consists of a version and a function
+// to apply when migration to that version is necessary.
+type Migration struct {
+	Version string
+	Apply   func(*gorm.DB) error
+}
+
+//
+// ShouldApply returns true if the .hub.db.version is lower than
+// this Migration's version.
+func (r *Migration) ShouldApply(db *gorm.DB) (apply bool, err error) {
+	setting := &model.Setting{}
+	result := db.Where("key", HubDBVersion).Find(setting)
+	if result.Error != nil {
+		err = liberr.Wrap(result.Error)
+		return
+	}
+
+	if setting.Value.String() < fmt.Sprintf("\"%s\"", r.Version) {
+		apply = true
+	}
+	return
+}
+
+//
+// All migrations in order.
+func All() []Migration {
+	return []Migration{
+		{Version: "v2.0.0", Apply: v200},
+		{Version: "v2.1.0", Apply: v210},
+	}
+}
+
+//
+// Migrate the hub by applying all necessary Migrations.
+func Migrate(migrations []Migration) (err error) {
```

Comment:
Would think the impl would be in migrate.go (or whatever) and not in the pkg.go.